### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by brianretford**

### CODE REVIEW

**Comment by earhart** [comment]
Maybe run yapf over this?

File: git_big/__init__.py [hunk 1]
```diff
@@ -12,4 +12,24 @@
...
 
-__version__ = '0.5.1'
+__version__ = '0.6.0'
+
+import os
+import os.path
+import platform
+import stat
+
+def setupForWindows():
```
Comment (flaub) [inline git_big/__init__.py]: Please use consistent naming: `setup_windows`

File: git_big/main.py [hunk 2]
```diff
@@ -362,8 +372,8 @@ def get(self, entry):
         cache_dir = os.path.dirname(entry.cache_path)
         if not os.path.exists(cache_dir):
             os.makedirs(cache_dir)
-        os.rename(pending_path, entry.cache_path)
```
Comment (flaub) [inline git_big/main.py]: Could you explain the reason for changing the order here? There is some refcounting that I thought I was exploiting here (on POSIX).

File: requirements-dev.txt [hunk 3]
```diff
@@ -1,42 +1,43 @@
...
-requests==2.18.4          # via bpython, requests-toolbelt, twine
-singledispatch==3.4.0.3   # via astroid, pylint
-six==1.11.0               # via astroid, bpython, packaging, pip-tools, pylint, singledispatch
-tqdm==4.19.4              # via twine
-twine==1.9.1
-urllib3==1.22             # via requests
-wcwidth==0.1.7            # via curtsies
-wrapt==1.10.11            # via astroid
-yapf==0.20.0
+#
```
Comment (flaub) [inline requirements-dev.txt]: Could you convert all the EOLs here? I doubt this whole file has changed.

File: setup.py [hunk 4]
```diff
@@ -30,7 +30,10 @@
         'apache-libcloud==1.5.0',
         'boto',
         'click',
+        'lockfile',
```
Comment (flaub) [inline setup.py]: This should only be a requirement for development.

File: setup.py [hunk 5]
```diff
@@ -40,11 +43,14 @@
...
         "Operating System :: Unix",
         "Operating System :: MacOS :: MacOS X",
+        "Operating System :: Microsoft :: Windows :: Windows 10"
         "Programming Language :: Python :: 2.7",
+        "Programming Language :: Python :: 3.6",
         "Topic :: Software Development :: Libraries",
         "Topic :: Software Development :: Version Control",
         "Topic :: Utilities",
     ],
+    scripts=['git-big-windows-setup'],
```
Comment (flaub) [inline setup.py]: Could we add this as a new command to the existing git-big commands so that it shows up in the help?

File: tests/test_pull.py [hunk 6]
```diff
@@ -58,9 +58,6 @@ def test_fresh_clone(env):
     assert islink(join(clone.repo_dir, 'foo'))
     assert islink(join(clone.repo_dir, 'bar'))
 
-    assert not isfile(join(clone.repo_dir, 'foo'))
```
Comment (flaub) [inline tests/test_pull.py]: I'm not sure these should go away.

File: tests/test_pull.py [hunk 7]
```diff
@@ -170,13 +167,13 @@ def test_pull_file(env):
     check_output(['git', 'big', 'pull', 'foo'])
 
     assert isfile(join(clone.repo_dir, 'foo'))
-    assert not isfile(join(clone.repo_dir, 'bar'))
```
Comment (flaub) [inline tests/test_pull.py]: When we pull a single file here, we really want to be sure we didn't accidently pull the other file, which is why we check for `not isfile`. We shouldn't allow a link to be formed in this case.

**Pull request merged**